docs/source/pickling_cuml_models.ipynb

```diff
@@ -0,0 +1,781 @@
+       "         5.24553   , -1.6519713 ,  6.152376  , -6.9043674 , -9.641573  ],\n",
+       "       [-4.6595864 , -9.578357  ,  6.6512713 ,  4.4457498 ,  2.1801045 ,\n",
+       "         2.611769  ,  0.5923821 ,  6.249009  , -8.817254  , -0.41071406,\n",
+       "         9.815973  ,  7.5775385 ,  9.986206  , -5.858885  , -1.2428757 ,\n",
+       "        -2.5559528 , -1.0446941 , -5.252986  , -9.3230715 ,  4.621556  ,\n",
+       "        -0.1072749 , -3.930478  ,  6.1892705 , -7.415579  ,  5.6697273 ,\n",
+       "        -8.56471   , -7.5154557 , -5.5349684 ,  4.859372  ,  2.5603094 ],\n",
+       "       [ 4.811571  ,  8.419027  , -9.238327  ,  9.380197  ,  8.494089  ,\n",
+       "        -1.0680069 ,  3.358382  , -7.799151  , -0.5922657 ,  0.2752325 ,\n",
+       "         5.511196  , -4.120103  ,  4.3025985 , -2.838152  ,  3.6216238 ,\n",
+       "        -4.170197  , -3.6244812 ,  6.208701  , -6.9180326 , -1.0881603 ,\n",
+       "        -5.8366294 ,  2.191313  , -3.8584485 , -1.7027951 , -5.3133397 ,\n",
+       "         7.5754147 ,  2.9267452 ,  8.543089  ,  1.5802664 ,  1.0812526 ]],\n",
+       "      dtype=float32)"
+      ]
+     },
+     "execution_count": 13,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "dist_model.cluster_centers_"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If an attribute doesn't exist, we get an error"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Failed\n"
+     ]
+    }
+   ],
+   "source": [
+    "try:\n",
+    "    dist_model.cluster_centers_fail_\n",
+    "except AttributeError:\n",
+    "    print(\"Failed\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The MG subclass also provides the ability to treat attributes differently when they don't make sense in the distributed context. \n",
+    "\n",
+    "For example, calling `combined_model.labels_` would only reference the labels for one partition. Collecting the labels assigned to the training inputs onto a single GPU will not scale, so this attribute gets blanked out in the MG version"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "None\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(dist_model.labels_)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If needed, the labels can be computed again"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "y_hat = dist_model.predict(X).compute().get()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Visualize Clusters"
```

Comment:
Maybe we get rid of the visualize clusters part here? I think if we want to keep the focus on pickling, this might be extending the notebook too much.